src/queue.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+
+export default class Queue {
+  constructor(initialValues=[]) {
+    this.elements = initialValues
+    this.first = initialValues[0]
+    this.last = initialValues.length
+  }
+    enqueue(value){
+      this.elements[this.last] = value
+      this.last++
+    }
+
+    dequeue(){
+      if(this.elements.length === 0 ){
+        return null
+      }
+      let result = this.elements.slice(0, 1)
+      this.elements.splice(0,1)
+      return result
+    }
+
+    front(){
+      if(this.elements.length === 0){
+        return null
+      } else {
+        return this.elements[0]
+      }
+    }
+
+    back(){
+      if(this.elements.length === 0){
```

Comment:
Cleaner implementation:

if (this.elements.length) {
   return this.elements.slice(-1)[0]
}

return null
```